Source/WebKit/webpushd/PushClientConnection.mm

```diff
@@ -102,13 +107,38 @@
         if (!m_hostAppAuditToken)
             m_hostAppCodeSigningIdentifier = String();
         else
-            m_hostAppCodeSigningIdentifier = WebKit::codeSigningIdentifier(*m_hostAppAuditToken);
+            m_hostAppCodeSigningIdentifier = bundleIdentifierFromAuditToken(*m_hostAppAuditToken);
 #endif
     }
 
     return *m_hostAppCodeSigningIdentifier;
 }
 
+String ClientConnection::bundleIdentifierFromAuditToken(audit_token_t audit_token)
+{
+#if PLATFORM(MAC)
+    LSSessionID sessionID = (LSSessionID)audit_token_to_asid(audit_token);
+    CFTypeRef auditTokenDataRef = CFDataCreate(kCFAllocatorDefault, (const UInt8 *)(&audit_token), sizeof(audit_token));
+    CFStringRef auditTokenKey = CFStringCreateCopy(kCFAllocatorDefault, _kLSAuditTokenKey);
+    CFArrayRef matchingAppsRef = _LSCopyMatchingApplicationsWithItems(sessionID, 1, (CFTypeRef *)(&auditTokenKey), &auditTokenDataRef);
+    CFRelease(auditTokenDataRef);
+    CFRelease(auditTokenKey);
```

Comment:
WebKit generally prefers smart pointers to avoid manual retain/release counts.
If this wouldn't work, e.g.:
auto auditTokenDataRef = adoptCF(CFDataCreate(kCFAllocatorDefault, (const UInt8 *)(&audit_token), sizeof(audit_token)));
Then this would:
RetainPtr<CFDataRef> auditTokenDataRef = adoptCF(CFDataCreate(kCFAllocatorDefault, (const UInt8 *)(&audit_token), sizeof(audit_token)));